src/main/asciidoc/repositories-scrolling.adoc

```diff
@@ -0,0 +1,83 @@
+[[repositories.scrolling]]
+==== Scrolling
+
+Scrolling is a more fine-grained approach to iterate through larger results set chunks.
+Scrolling consists of a stable sort, a scroll type (Offset- or Keyset-based scrolling) and result limiting.
+You can define simple sorting expressions by using property names and define static result limiting using the <<repositories.limit-query-result,`Top` or `First` keyword>> through query derivation.
+You can concatenate expressions to collect multiple criteria into one expression.
+
+Scroll queries return a `Scroll<T>` that allows obtaining the scroll position to resume to obtain the next `Scroll<T>` until your application has consumed the entire query result.
+Similar to consuming a Java `Iterator<List<…>>` by obtaining the next batch of results, query result scrolling lets you access the next `ScrollPosition`  through `Scroll.lastScrollPosition()`.
+
+[[repositories.scrolling.offset]]
+===== Scrolling using Offset
+
+Offset scrolling uses similar to pagination, an Offset counter to skip a number of results and let the data source only return results beginning at the given Offset.
+This simple mechanism avoids large results being sent to the client application.
+However, most databases require materializing the full query result before your server can return the results.
+
+.Using `OffsetScrollPosition` with Repository Query Methods
+====
+[source,java]
+----
+interface UserRepository extends Repository<User, Long> {
+
+  Scroll<User> findFirst10ByLastnameOrderByFirstname(String lastname, OffsetScrollPosition position);
+}
+
+Scroll<User> users = repository.findFirst10ByLastnameOrderByFirstname("Doe", OffsetScrollPosition.initial());
+
+do {
+
+  for (User u : users) {
+    // consume the user
+  }
+
+  // obtain the next Scroll
+  users = repository.findFirst10ByLastnameOrderByFirstname("Doe", users.lastScrollPosition());
```

Comment:
When reading this, I had a hard time to find out what the `ScrollPosition` actually does. Is it including, like give me everything starting (inclusive) at a certain position. Or is it excluding, like return the data and start after the position. 
My point here is, should we call it `lastScrollPosition` or would it better be `nextScrollPosition` which is also not quite accurate. Does it maybe make sense to have a scroll position that has `startAt` and `startAfter`?
Does this read any better `ScrollPosition.after(users.lastScrollPosition())` or is it even more confusing?